lib/transformer.cc

```diff
@@ -224,4 +224,68 @@ Canvas *LargeSquare64x64Transformer::Transform(Canvas *output) {
   return canvas_;
 }
 
+/***********************************/
+/* Half Transformer Canvas */
+/***********************************/
+class HalfTransformer::TransformCanvas : public Canvas {
+public:
+  TransformCanvas() : delegatee_(NULL) {}
+
+  void SetDelegatee(Canvas* delegatee);
+
+  virtual void Clear();
+  virtual void Fill(uint8_t red, uint8_t green, uint8_t blue);
+  virtual int width() const;
+  virtual int height() const;
+  virtual void SetPixel(int x, int y, uint8_t red, uint8_t green, uint8_t blue);
+
+private:
+  Canvas *delegatee_;
+};
+
+void HalfTransformer::TransformCanvas::SetDelegatee(Canvas* delegatee) {
+  // Our assumptions of the underlying geometry:
+  assert(delegatee->height() % 2 == 0);
+
+  delegatee_ = delegatee;
+}
+
+void HalfTransformer::TransformCanvas::Clear() {
+  delegatee_->Clear();
+}
+
+void HalfTransformer::TransformCanvas::Fill(uint8_t red, uint8_t green, uint8_t blue) {
+  delegatee_->Fill(red, green, blue);
+}
+
+int HalfTransformer::TransformCanvas::width() const {
+  return delegatee_->width();
+}
+
+int HalfTransformer::TransformCanvas::height() const {
+  return delegatee_->height() / 2;
+}
+
+void HalfTransformer::TransformCanvas::SetPixel(int x, int y, uint8_t red, uint8_t green, uint8_t blue) {
+  delegatee_->SetPixel(x, y, red, green, blue);
```

Comment:
mmh, this looks like it will only work with one chain.
If you have one chain (e.g. your 16 pixel high display), then this works.

If you have three of these with `-P3`, then you actually need to do it a little more complicated by mapping 
  * 0..15 -> 0..15  (what you are doing now)
  * 16..31 -> 32..47 (the second panel, which 'thinks' it is from 32..64, but is actually just starting at line 16..)
  * 32..47 -> 48..63 (dito, same sentiment).